sig-scalability/slos/pod_startup_latency.md

```diff
@@ -0,0 +1,47 @@
+## Pod startup latency SLI/SLO details
+
+### User stories
+- As a user of vanilla Kubernetes, I want some guarantee how quickly my pods
+will be started.
+
+### Other notes
+- Only schedulable and stateless pods contribute to the SLI:
+  - If there is no space in the cluster to place the pod, there is not much
+    we can do about it (it is task for Cluster Autoscaler which should have
+    separate SLIs/SLOs).
+  - If placing a pod requires preempting other pods, that may heavily depend
+    on the application (e.g. on their graceful termination period). We don't
+    want that to contribute to this SLI.
+  - Mounting disks required by non-stateless pods may potentially also require
```

Comment:
I need to think if there are more scenarios under which the SLI may undesirably be affected. But maybe we don't need to block this PR on that.

A thought - it might be easier to define what scenario we "include" instead of explicitly providing all scenarios we "exclude"?

Reply:
There are at least following other "exclude" scenarios I could think of in the meanwhile which might affect the latency:

- too many secrets/configmaps used by the pod
- too big secrets/configmaps?
- pod with complicated scheduling (anti-affinity, etc)

My feeling is we might be able to express our include/exclude scenarios in a more succinct way (sth like "stateless and schedulable pods that don't exercise custom scheduling/setup features that can cause increased delays in running the pod (e.g antiaffinity. volume mounts)"). I know that isn't perfect, but just giving the idea. Feel free to disagree.

Reply:
1. too many/too big secrets/configmaps - if we have numbers we may consider changing that; I don't want to provide something that isn't very clear.
2. scheduling features - that isn't something that should be excluded from SLI; it may be TODO to improve it, but they should be covered by SLI

Reply:
Ok - that sounds reasonable to me.